src/utilities/theme/utils.ts

```diff
@@ -1,115 +1,779 @@
+    ) as HSLColor),
+  );
+
+  const {
+    surface = DefaultColor.Surface,
+    onSurface = DefaultColor.OnSurface,
+    interactive = DefaultColor.Interactive,
+    branded = DefaultColor.Branded,
+    critical = DefaultColor.Critical,
+    warning = DefaultColor.Warning,
+    highlight = DefaultColor.Highlight,
+    success = DefaultColor.Success,
+  } = colors;
 
-const lightenToString: (
-  color: HSLColor | string,
-  lightness: number,
-  saturation: number,
-) => string = compose(
-  hslToString,
-  createLightColor,
-);
-
-export function setTextColor(
-  name: string,
-  variant: ThemeVariant = 'dark',
-): string[] {
-  if (variant === 'light') {
-    return [name, tokens.colorInk];
+  return customPropertiesTransformer({
+    ...surfaceColors(surface),
+    ...onSurfaceColors(onSurface),
+    ...interactiveColors(interactive),
+    ...interactiveNeutralColors(onSurface),
+    ...brandedColors(branded),
+    ...criticalColors(critical),
+    ...warningColors(warning),
+    ...highlightColors(highlight),
+    ...successColors(success),
+  });
+
+  function surfaceColors(baseColor: string): CSSProperties {
+    const hslBaseColor = colorToHsla(baseColor) as HSLColor;
+    const {hue, saturation} = hslBaseColor;
+
+    return {
+      surface: hslToString(hslBaseColor),
+      surfaceBackground: hslToString({
+        hue,
+        saturation,
+        // TODO: Should the dark version be 0 lightness to save battery?
+        lightness: isLightTheme ? 98 : 7,
+      }),
+      surfaceForeground: hslToString({
+        hue,
+        saturation,
+        lightness: isLightTheme ? 100 : 13,
+      }),
+      surfaceForegroundSubdued: hslToString({
+        hue,
+        saturation,
+        lightness: isLightTheme ? 90 : 10,
+      }),
+      surfaceOpposite: hslToString({
+        hue,
+        saturation,
+        lightness: isLightTheme ? 0 : 100,
+      }),
+    };
   }
 
-  return [name, tokens.colorWhite];
-}
+  function onSurfaceColors(baseColor: string): CSSProperties {
+    const hslBaseColor = colorToHsla(baseColor) as HSLColor;
+    const {hue, saturation} = hslBaseColor;
 
-export function setTheme(
-  color: string | HSLColor,
-  baseName: string,
-  key: string,
-  variant: 'light' | 'dark',
-): string[][] {
-  const colorPairs = [];
-  switch (variant) {
-    case 'light':
-      colorPairs.push(
-        setTextColor(constructColorName(baseName, null, 'color'), 'light'),
-      );
-
-      colorPairs.push([
-        constructColorName(baseName, key, 'lighter'),
-        lightenToString(color, 7, -10),
-      ]);
-
-      break;
-    case 'dark':
-      colorPairs.push(
-        setTextColor(constructColorName(baseName, null, 'color'), 'dark'),
-      );
-
-      colorPairs.push([
-        constructColorName(baseName, key, 'lighter'),
-        lightenToString(color, 15, 15),
-      ]);
-
-      break;
-    default:
+    return {
+      onSurface: hslToString(hslBaseColor),
+      actionOnDark: hslToString({
```

Comment:
I think this makes sense. I had been reluctant to DRY this code up since we might end up tinkering with `saturation`, but I think it makes sense to handle that case if/when we come to it.